backend/controllers/passController.js

```diff
@@ -0,0 +1,71 @@
+const User = require("../model/user.model")
+const Pass = require("../model/pass.model")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+
+//@desc Generate a pass by user
+//@route POST /users/generatePass
+const generatePass = async (req, res, next) => {
+  try {
+    const { name, email, phone, duration, start } = req.body
+
+    const newPass = await Pass.create({
+      name,
+      phone,
+      email,
+      duration,
+      start,
+      generatedUserId: req.user.id,
+      userName: req.user.name,
+    })
+    if (newPass) {
+      res.status(201).json({
+        success: true,
+        newPass: newPass,
+        message: "Pass generated successfully",
+      })
+    }
+  } catch (err) {
+    console.log(err)
+    next(err)
+  }
+}
+
+//@desc View all passes generated by a user
+//@route Get /users/viewPasses
+const viewPasses = async (req, res, next) => {
+  try {
+    const passes = await Pass.find({ generatedUserId: req.user.id })
+
+    if (passes.length > 0) {
```

Comment:
For that I need to add the following condition so that it returns empty array, otherwise no valid response will be recorded
`else if (passes.length === 0) {
      res.status(201).json({
        passes: passes,
      })
    }`

Reply:
What if you remove the current condition(that checks the size)? Will that create any issue?